Wasm/Wast/AST.lean

```diff
@@ -0,0 +1,165 @@
+import Wasm.Wast.BitSize
+import Wasm.Wast.Num
+
+open Wasm.Wast.Num.Num.Int
+open Wasm.Wast.Num.Num.Float
+open Wasm.Wast.Num.Uni
+
+namespace Wasm.Wast.AST
+section AST
+
+
+namespace Type'
+
+inductive Type' where
+  | f : BitSize → Type'
+  | i : BitSize → Type'
+  -- | v : BitSizeSIMD → Type'
+  deriving BEq
+
+instance : ToString Type' where
+  toString x := match x with
+  | .f y => "(Type'.f " ++ toString y ++ ")"
+  | .i y => "(Type'.i " ++ toString y ++ ")"
+  -- | .v y => "(Type'.v " ++ toString y ++ ")"
+
+def numUniType : NumUniT → Type'
+  | .i x => .i x.bs
+  | .f x => .f x.bs
+
+end Type'
+open Type'
+
+
+namespace Local
+
+structure Local where
+  index : Nat
+  name : Option String
+  type : Type' -- TODO: We need to pack lists with different related types'. For that we need something cooler than List, but since we're just coding now, we'll do it later.
+  deriving BEq
+
+instance : ToString Local where
+  toString x := s!"(Local.mk {x.index} {x.type})"
+
+end Local
+open Local
+
+
+namespace Get
+
+inductive Get (x : Type') where
+| from_stack
+| by_name : Local → Get x
+| by_index : Local → Get x
+
+instance : ToString (Get α) where
+  toString x := "(" ++ (
+    match x with
+    | .from_stack => "Get.from_stack"
+    | .by_name n => "Get.by_name " ++ toString n
+    | .by_index i => "Get.by_index " ++ toString i
+  ) ++ " : Get " ++ toString α ++ ")"
+
+end Get
+open Get
+
+
+/- TODO: Instructions are rigid WAT objects. If we choose to only support
+S-Expressions at this point, we don't need this concept. -/
+namespace Instruction
+end Instruction
+
+
+namespace Operation
+
+mutual
+  -- Sadge
```

Comment:
Probably we should mention here that were are sad because there is no way to mut rec dependently typed objects with less dependent ones.